bin/improver-spot-extract-new

```diff
@@ -0,0 +1,161 @@
+        "processed, the lapse rates will be used to adjust the temperatures "
+        "to better represent each spot site's altitude.")
+
+    meta_group = parser.add_argument_group("Metadata")
+    meta_group.add_argument(
+        "--grid_metadata_identifier", default="mosg__grid",
+        help="A string to identify attributes from the input netCDF files that"
+        " should be compared to ensure that the data is compatible. Spot"
+        " data works using grid indices, so it is important that the grids"
+        " are matching or the data extracted may not match the location of the"
+        " spot data sites.")
+
+    meta_group.add_argument(
+        "--json_file", metavar="JSON_FILE", default=None,
+        help="If provided, this JSON file can be used to modify the metadata "
+        "of the returned netCDF file. Defaults to None.")
+
+    args = parser.parse_args()
+    neighbour_cube = load_cube(args.neighbour_filepath)
+    diagnostic_cube = load_cube(args.diagnostic_filepath)
+
+    neighbour_selection_method = NeighbourSelection(
+        land_constraint=args.land_constraint,
+        minimum_dz=args.minimum_dz).neighbour_finding_method_name()
+
+    plugin = SpotExtraction(
+        neighbour_selection_method=neighbour_selection_method,
+        grid_metadata_identifier=args.grid_metadata_identifier)
+    result = plugin.process(neighbour_cube, diagnostic_cube)
+
+    # Check whether a lapse rate cube has been provided and we are dealing with
+    # temperature data.
+    if (args.temperature_lapse_rate_filepath and
+            diagnostic_cube.name() == "air_temperature"):
+
+        lapse_rate_cube = load_cube(args.temperature_lapse_rate_filepath)
+        try:
+            lapse_rate_height_coord = lapse_rate_cube.coord("height")
+            lapse_rate_height, = lapse_rate_height_coord.points
+        except (ValueError, CoordinateNotFoundError):
+            msg = ("Lapse rate cube does not contain a single valued height "
+                   "coordinate. This is required to ensure it is applied to "
+                   "equivalent temperature data.")
+            raise ValueError(msg)
+
+        # Check the height of the temperature data  matches that used to
+        # calculate the lapse rates. If so, adjust temperatures using the lapse
+        # rate values.
+        if diagnostic_cube.coord("height") == lapse_rate_height_coord:
+            plugin = SpotLapseRateAdjust(
+                args.grid_metadata_identifier,
+                neighbour_selection_method=neighbour_selection_method)
+            result = plugin.process(result, neighbour_cube, lapse_rate_cube)
+        else:
+            msg = ("A lapse rate cube was provided, but the height of "
+                   "the temperature data does not match that of the data used "
+                   "to calculate the lapse rates. As such the temperatures "
+                   "were not adjusted with the lapse rates.")
+            warnings.warn(msg)
```

Comment:
I'm not sure why we're allowing the user to pass in a temperature-lapse-rate cube for files that do not contain "air_temperature". Could you add either a warning or an error if the lapse-rate is provided but not used?

Reply:
I've added a warning to state that it has been provided but not used. CLI test 12 has been added to cover this.